build.assets/tooling/cmd/resource-ref-generator/reference/reference_test.go

```diff
@@ -0,0 +1,358 @@
+					Name:    "AbsentType",
+				},
+			},
+			expected: false,
+		},
+		{
+			description: "field from the same package",
+			src: `package testpkg
+type MyStruct struct{
+  Metadata       types.Metadata
+  AlsoMetadata   Metadata
+  Name           otherpkg.TypeName
+}
+`,
+			requiredFields: []TypeInfo{
+				{
+					Package: "testpkg",
+					Name:    "Metadata",
+				},
+			},
+			expected: true,
+		},
+		{
+			description: "one required type with long path from a separate package",
+			src: `package testpkg
+type MyStruct struct{
+  Metadata       types.Metadata
+  AlsoMetadata   Metadata
+  Name           otherpkg.TypeName
+}
+`,
+			requiredFields: []TypeInfo{
+				{
+					Package: "long/package/path/types",
+					Name:    "Metadata",
+				},
+			},
+			expected: true,
+		},
+		{
+			description: "one required type with long path from the current  package",
+			src: `package testpkg
+type MyStruct struct{
+  Metadata       types.Metadata
+  AlsoMetadata   Metadata
+  Name           otherpkg.TypeName
+}
+`,
+			requiredFields: []TypeInfo{
+				{
+					Package: "long/package/path/testpkg",
+					Name:    "Metadata",
+				},
+			},
+			expected: true,
+		},
+		{
+			description: "excluded type",
+			src: `package testpkg
+type MyStruct struct{
+  Metadata       types.Metadata
+  AlsoMetadata   Metadata
+  Name           otherpkg.TypeName
+}
+`,
+			requiredFields: []TypeInfo{
+				{
+					Package: "testpkg",
+					Name:    "Metadata",
+				},
+			},
+			excludedResources: []TypeInfo{
+				{
+					Package: "testpkg",
+					Name:    "MyStruct",
+				},
+			},
+			expected: false,
+		},
+	}
+
+	for _, c := range cases {
+		// Parse the fixture as an AST node so we can use it in shouldProcess.
+		fset := token.NewFileSet()
+		d, err := parser.ParseFile(fset,
+			"myfile.go",
+			c.src,
+			parser.ParseComments,
+		)
+		if err != nil {
+			t.Fatalf("test fixture contains invalid Go source: %v\n", err)
+		}
+
+		if len(d.Decls) != 1 {
+			t.Fatal("the source fixture must contain a single *ast.GenDec (this is a problem with the test)")
+
+		}
+
+		l, ok := d.Decls[0].(*ast.GenDecl)
+		if !ok {
+			t.Fatal("the source fixture must contain a single *ast.GenDec (this is a problem with the test)")
+
+		}
+
+		t.Run(c.description, func(t *testing.T) {
+			assert.Equal(t, c.expected, shouldProcess(resource.DeclarationInfo{
+				FilePath:    "myfile.go",
+				Decl:        l,
+				PackageName: "testpkg",
+			}, c.requiredFields, c.excludedResources))
+		})
+	}
+}
+
+// This test reads the golden files at the destination directory and compares
+// the generated resource reference docs with them. To regenerate the golden
+// files, delete the destination directory (reference/testdata/dest) and run the
+// test again.
+func TestGenerate(t *testing.T) {
+	// Define paths based on relative paths from this Go source file to
+	// guarantee consistent results regardless of where a user runs "go test".
+	_, callerPath, _, _ := runtime.Caller(0)
```

Comment:
You shouldn't need to do this. `go test` guarantees that the current working directory will be the directory containing the test (`reference` in this case).